app.py

```diff
@@ -64,23 +62,89 @@ def download_chapter(url):
-        with open('{}.mp3'.format(fname), 'wb') as outfile:
-            outfile.write(download_chapter(url))
-
-    print('All done.')
+    #set up command-line arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-o', '--outputDir',
+                        help='Directory the AudioBook will be dowloaded to. Default: <AudioBookName>',
+                        metavar='OutputDirectory',
+                        type=str)
+    parser.add_argument('audioBookURL',
+                        help='a URL of a multi-part audio book',
+                        metavar='URL')
+    parser.add_argument('-w', '--overwrite',
+                        help='overwrite existing directory without asking',
+                        action='store_true',
+                        default=False)
+    parser.add_argument('-c', '--cleanup',
+                        help='delete outputDir in case of errors',
+                        action='store_true',
+                        default=False)
+
+    #parse arguments
+    args = parser.parse_args()
+
+    #default value for output dir is the book name
+    if args.outputDir is None:
+        args.outputDir = args.audioBookURL.split('/')[-1]
+
+    fullPathDir = os.path.abspath(args.outputDir)
+
+    #Check outputDir
+    if os.path.exists(fullPathDir):
+        #outputDir Exists
+        if not os.path.isdir(fullPathDir):
+            #outputPath is not a directory
+            print("{} exists, and is not a directory!".format(fullPathDir))
+            exit(1)
+        elif os.listdir(fullPathDir) and not args.overwrite:
+            #outputPath is a directory, is not empty and overwrite flag is off
+            if not yes_or_no('Directory <{}> exists. Overwrite?'.format(fullPathDir)):
+                #Prompt for overwrite returned NO
+                sys.exit(1)
+            else:
+                #Prompt for overwrite returned YES
+                print("overwriting files in <{}>\n".format(fullPathDir))
+    else:
+      #outputDir does not exist - create directory
+      print("Creating directory <{}>".format(fullPathDir))
+      os.makedirs(fullPathDir)
+
+    print("\nDownloading audiobook from {} to <{}>\n".format(args.audioBookURL,
+                                                           fullPathDir))
+
+    try:
+        with open_browser(args.audioBookURL) as browser:
+              book_id = get_book_id(browser.page_source)
+              playlist = get_playist(browser, book_id)
+
+        for url, fname in playlist:
+            print('Downloading chapter "{}"'.format(fname))
+            with open('{}.mp3'.format(os.path.join(fullPathDir,fname)), 'wb') as outfile:
+                outfile.write(download_chapter(url))
+
+        print('All done\n')
+
+    except Exception as e:
+      #remove previously created directory
+      if (args.cleanup or
+          (os.listdir(fullPathDir) and
+           yes_or_no('Error encountered. Remove Audiobook directory?')
```

Comment:
Look at this [one](http://click.palletsprojects.com/en/7.x/prompts/).

Reply:
Will do, as I move everything to Click